pkg/template/registry/templateinstance/strategy.go

```diff
@@ -104,22 +104,16 @@ func (s *templateInstanceStrategy) ValidateUpdate(ctx apirequest.Context, obj, o
 	// place where this happens is in the garbage collector, which uses
 	// Unstructureds via the dynamic client.
 
-	objcopy, err := kapi.Scheme.DeepCopy(obj)
-	if err != nil {
-		return field.ErrorList{field.InternalError(field.NewPath(""), err)}
-	}
-	templateInstance := objcopy.(*templateapi.TemplateInstance)
+	templateInstanceCopy := obj.DeepCopyObject()
+	templateInstance := templateInstanceCopy.(*templateapi.TemplateInstance)
 
 	errs := runtime.DecodeList(templateInstance.Spec.Template.Objects, unstructured.UnstructuredJSONScheme)
 	if len(errs) != 0 {
 		return field.ErrorList{field.InternalError(field.NewPath(""), kutilerrors.NewAggregate(errs))}
 	}
 
-	oldcopy, err := kapi.Scheme.DeepCopy(old)
-	if err != nil {
-		return field.ErrorList{field.InternalError(field.NewPath(""), err)}
-	}
-	oldTemplateInstance := oldcopy.(*templateapi.TemplateInstance)
+	oldTemplateInstanceCopy := old.DeepCopyObject()
```

Comment:
can old be nil?

Reply:
i don't think it can be nil, but I will double check